pkgs/development/compilers/llvm/4.0/libc++/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetch, cmake, llvm, libcxxabi, fixDarwinDylibNames, version }:
+
+stdenv.mkDerivation rec {
+  name = "libc++-${version}";
+
+  src = fetch "libcxx" "130clvfffqml8hbnlvr596nfjk18n6ifxab27xl66nxhq99wccfn";
+
+  postUnpack = ''
+    unpackFile ${libcxxabi.src}
+    export LIBCXXABI_INCLUDE_DIR="$PWD/$(ls -d libcxxabi-${version}*)/include"
+  '';
+
+  preConfigure = ''
+    # Get headers from the cxxabi source so we can see private headers not installed by the cxxabi package
+    cmakeFlagsArray=($cmakeFlagsArray -DLIBCXX_CXX_ABI_INCLUDE_PATHS="$LIBCXXABI_INCLUDE_DIR")
+  '';
+
+  patches = lib.optional stdenv.isDarwin ./darwin.patch;
```

Comment:
Pushed commit that tries the substituteInPlace approach, let me know if that works for you-- looking at [the 4.0 version of this file](https://github.com/llvm-mirror/libcxx/blob/release_40/lib/CMakeLists.txt#L131) I think this will accomplish what the patch did previously.

As a minor note: at least on Linux it seems the /usr/lib/libSystem path gets rewritten to /var/empty/libSystem which definitely seems wrong.  Or maybe it doesn't matter :).